internal/config/config.go

```diff
@@ -556,11 +643,59 @@ func validInstanceTypeForProvider(insType string, acceptNonCVM bool, provider cl
 			}
 		}
 		return false
+	case cloudprovider.GCP:
+		for _, instanceType := range instancetypes.GCPInstanceTypes {
+			if insType == instanceType {
+				return true
+			}
+		}
+		return false
 	default:
 		return false
 	}
 }
 
+// checkIfAWSInstanceTypeIsValid checks if an AWS instance type passed as user input is in one of the instance families supporting NitroTPM-
+func checkIfAWSInstanceTypeIsValid(userInput string) bool {
+	// Check if user or code does anything weird and tries to pass multiple strings as one
+	if strings.Contains(userInput, " ") {
+		return false
+	}
+	if strings.Contains(userInput, ",") {
+		return false
+	}
+	if strings.Contains(userInput, ";") {
+		return false
+	}
+
+	userInputLowercase := strings.ToLower(userInput)
+	splitInstanceType := strings.Split(userInputLowercase, ".")
+
+	if len(splitInstanceType) != 2 {
+		return false
+	}
+
+	userDefinedFamily := splitInstanceType[0]
+	userDefinedSize := splitInstanceType[1]
+
+	// Check if instace type has at least 4 vCPUs (= contains "xlarge" in its name)
+	hasEnoughVCPUs := strings.Contains(userDefinedSize, "xlarge")
+	if !hasEnoughVCPUs {
+		return false
+	}
+
+	// Now check if the user input is a supported family
+	// Note that we cannot directly use the family split from the Graviton check above, as some instances are directly specified by their full name and not just the family in general
+	for _, supportedFamily := range instancetypes.AWSSupportedInstanceFamilies {
+		supportedFamilyLowercase := strings.ToLower(supportedFamily)
```

Comment:
I think we actually need to check the casing here. `C5.xlarge` is not a valid instance type while `c5.xlarge` is.

Reply:
Should now be enforced by go-validator.